llvm/lib/CodeGen/ReplaceWithVeclib.cpp

```diff
@@ -100,20 +100,39 @@ static void replaceWithTLIFunction(IntrinsicInst *II, VFInfo &Info,
 static bool replaceWithCallToVeclib(const TargetLibraryInfo &TLI,
                                     IntrinsicInst *II) {
   assert(II != nullptr && "Intrinsic cannot be null");
-  // At the moment VFABI assumes the return type is always widened unless it is
```

Comment:
this comment was useful as it explains the why EC is computed in such a way please do not delete it

Reply:
I made the change to rely on using `isVectorIntrinsicWithScalarOpAtArg` as well because it seemed like the right thing to do.  If it ever returned true for the return index `-1`, that would contradict the comment.

However, since I see other paths do not use this for the return and this doesn't return true for index `-1` for any intrinsic, I can change this part back, since it would likely require other code changes if this assumption were to ever change.